network/src/transport.rs

```diff
@@ -60,20 +61,19 @@ impl From<u32> for ConnectionId {
 
 /// Generator of unique ConnectionIds.
 struct ConnectionIdGenerator {
-    next: ConnectionId,
+    ctr: AtomicU32,
 }
 
 impl ConnectionIdGenerator {
-    fn new() -> ConnectionIdGenerator {
+    const fn new() -> ConnectionIdGenerator {
         Self {
-            next: ConnectionId(0),
+            ctr: AtomicU32::new(0),
         }
     }
 
-    fn next(&mut self) -> ConnectionId {
-        let ret = self.next;
-        self.next = ConnectionId(ret.0.wrapping_add(1));
-        ret
+    fn next(&self) -> ConnectionId {
+        let next = self.ctr.fetch_add(1, Ordering::Relaxed);
```

Comment:
shouldn't this be SeqCst?

Reply:
we don't care what the ordering is, just that the increment is atomic and therefore returns a unique id (no torn reads/writes etc)

Reply:
I don't have a lot of understanding for this, but isn't the danger that you start a connection with connection id = X when you wanted X+1 because of some compiler/machine re-ordering?

Reply:
again, we just care that each call to `next` returns a unique id. the ordering relative to other atomics/instructions doesn't matter.